3d_armor/api.lua

```diff
@@ -258,21 +270,42 @@ armor.set_player_armor = function(self, player)
 				local value = def.groups["armor_"..attr] or 0
 				attributes[attr] = attributes[attr] + value
 			end
-			local mat = string.match(item, "%:.+_(.+)$")
-			if material.name then
-				if material.name == mat then
-					material.count = material.count + 1
+		end
+	end
+	-- The following code compares player worn armor items against requirements
+	-- of which armor pieces are needed to be worn to meet set bonus requirements
+	for loc,item in pairs(worn_armor) do
+		local item_mat = string.match(item, "%:.+_(.+)$")
+		for k,set_loc in pairs(armor.config.set_elements)do
+			if set_loc == loc then
+				if item_mat ~= nil then
```

Comment:
Shorter:
```Lua
local worn_key = item_mat and item_mat or "unknown"
-- use "worn_key" to index "set_worn" here
```

Reply:
@SmallJoker Thats very cool and saves quiet a bit of code, I don't know enough about lua but is there any need for item_mat twice? I tested using single and it works as expected but not sure if this is my lack of understanding?

Used this instead:
`local worn_key = item_mat or "unknown"`

Full new code at the moment:
```
	for loc,item in pairs(worn_armor) do
		local item_mat = string.match(item, "%:.+_(.+)$")
		local worn_key = item_mat or "unknown"		
		for k,set_loc in pairs(armor.config.set_elements)do		
			if set_loc == loc then
				if set_worn[worn_key] == nil then
					set_worn[worn_key] = 0
					set_worn[worn_key] = set_worn[worn_key] + 1
				else
					set_worn[worn_key] = set_worn[worn_key] + 1
				end
			end
		end
	end
```

I re-tested to confirm behaviour remains unchanged

Reply:
Oh. I did not notice that `item_mat` is the condition and key at the same time. Yes of course, `item_mat or "unknown"` works just fine.